apps/mobile/pushtracker/src/app/services/translation.service.ts

```diff
@@ -52,32 +63,45 @@ export class TranslationService extends Observable {
         const isMax = v === maxes[fName];
         return isMax && (!current || v > currentVersion);
       });
+
       const files = [];
       // do we need to download any language files?
       if (fileMetadatas && fileMetadatas.length) {
         for (let i = 0; i < fileMetadatas.length; i++) {
-          try {
-            const f = fileMetadatas[i];
-            this._logService.logBreadCrumb(
-              TranslationService.name,
-              `Downloading language file update ${f['_filename']} version ${f['_version']}`
-            );
-            const dl = await TranslationService.download(f);
-            files.push(dl);
-          } catch (err) {
+          const f = fileMetadatas[i];
+          this._logService.logBreadCrumb(
+            TranslationService.name,
+            `Downloading language file update ${f['_filename']} version ${f['_version']}`
+          );
+
+          const dl = await TranslationService.download(f).catch(err => {
             this._logService.logBreadCrumb(
               TranslationService.name,
-              'Could not download language files'
+              `Could not download language files: ${err.toString()}`
             );
-            // this._logService.logException(err);
-          }
+            // clear out the data - we couldn't save the file
+            delete this.currentVersions[f.name];
+          });
+
+          files.push(dl);
```

Comment:
what will the value of `dl` be here if the error handler is triggered I believe we should check the value of dl, yes?
```suggestion
          if (dl) files.push(dl);
```